### PR TITLE
Delete list route

### PR DESCRIPTION
**Pull request by LebkuchenCJ**
## Comments:
 - This PR added a deletion function / route to my lists
 - nothing more, nothing less

### CODE REVIEW

File: lib/models/list.js [hunk 1]
```diff
@@ -27,7 +27,17 @@ async function addList(name, creationDate, products) {
...
+  if (!list) {
+    return null;
+  }
+  return list;
+}
 
 exports.readLists = readLists;
 exports.readList = readList;
 exports.addList = addList;
+exports.deleteList = deleteList;
```
Comment (mschmauder-hub) [inline lib/models/list.js]: You can export this as a module 
```suggestion
module.exports = {readLists, readList, addList, deleteList};
```

**Review by mschmauder-hub — COMMENTED** [review]
gj!

File: client/src/components/ListDeleter.js [hunk 2]
```diff
@@ -2,13 +2,20 @@ import React from "react";
...
     <Wrapper>
       <h4>Do you want to delete this list?</h4>
       <div>
         <button onClick={() => onCancel(false)}>Cancel</button>
-        <button onClick={() => onCancel(false)}>Proceed</button>
+        <button
+          onClick={() => {
+            onDeleteConfirm(id);
+            onCancel(false);
+          }}
```
Comment (lmachens) [inline client/src/components/ListDeleter.js]: Why is it necessary to call `onCancel`?
Shouldn't it enough to just call the confirmation?

File: client/src/pages/Home.js [hunk 3]
```diff
@@ -39,7 +43,7 @@ function Home() {
               key={list._id}
               list={list}
               href={`/home/${list._id}`}
-              onDelete={setDeleterOverlay}
+              onDeleteConfirm={handleDelete}
```
Comment (lmachens) [inline client/src/pages/Home.js]: Think about this: (your other components will be more simple).

```suggestion
              onDeleteConfirm={() => handleDelete(list._id)}
```

**Review by lmachens — COMMENTED** [review]
gj

File: client/src/components/ListDeleter.js (hunk 2 above)
Comment (LebkuchenCJ) [reply client/src/components/ListDeleter.js]: onCancel will close the overlay window

File: client/src/pages/Home.js (hunk 3 above)
Comment (LebkuchenCJ) [reply client/src/pages/Home.js]: Tried it before, and it wouldn't close my overlay, but now it's working :/ Thanks for the input, I already changed it and shortend my components

**Pull request merged**